Tests/XMLCoderTests/CharacterDataDecodingTestCase.swift

```diff
@@ -29,4 +29,42 @@ class CharacterDataDecodingTestCase: DecodingTestCase {
         // then
         XCTAssertEqual(attr, Attribute(name: "Xpos", value: "0"))
     }
+
+	func testItDecodesCharDataWhenNested() {
+		// given
+		struct Item: Decodable, Equatable {
+			enum CodingKeys: String, CodingKey {
+				case id = "ID"
+				case creator = "Creator"
+				case children = "attribute"
+			}
+
+			let id: String
+			let creator: String
+			let children: [Attribute]?
+		}
+
+		struct Attribute: Decodable, Equatable {
+			enum CodingKeys: String, CodingKey {
+				case name
+				case value = "#text"
+			}
+
+			let name: String
+			let value: String?
+		}
+
+		let xml = """
+			<item ID="1542637462" Creator="Francisco Moya">
+			<attribute name="Xpos">0</attribute>
+			</item>
+			"""
+
+		// when
+		let result: Item = try! decode(xml)
+
+		// then
+		let expected = Item(id: "1542637462", creator: "Francisco Moya", children: nil)
```

Comment:
I'm a bit confused why `children` gets a `nil` value here, isn't it supposed to be `Attribute(name: "Xpos", value: "0")`?

Reply:
Hey Max! Thanks—by mistake, I pushed some things to this PR that should have gone elsewhere. My last intended commit was ‘Complete if #availables’. I’ll revert as appropriate and fix the automated formatting. Thanks!